babel/messages/pofile.py

```diff
@@ -521,14 +522,21 @@ def _write(text):
             text = text.encode(catalog.charset, 'backslashreplace')
         fileobj.write(text)
 
+<<<<<<< HEAD
+    def _write_comment(comment, prefix='':
+=======
     def _write_comment(comment, prefix=''):
-        # xgettext always wraps comments even if --no-wrap is passed;
-        # provide the same behaviour
-        if width and width > 0:
-            _width = width
-        else:
-            _width = 76
-        for line in wraptext(comment, _width):
+>>>>>>> 67aa652a1c10bfdc92587cdae87fdab65d633b23
+        # NEVER wrap comments, this observation: "xgettext always wraps comments even if --no-wrap is passed;" is FALSE. There seemed to be a bug in the xgettext code, because wrapping doesn't always occur
+        # Make sure comments are unique and sorted alphabetically so locations can be easily searched and identify
+        has_comment = (bool(comment) and len(comment) > 0)
+        if not has_comment:
+            return
+
+        is_list_type = isinstance(comment, list)
+        comment_list = (list(comment) if not is_list_type else comment)
+        comment_list.sort()
```

Comment:
Can you explain what's happening here? Is it possible this reorders multiline comments into the wrong order?

Reply:
As I run the code in tests, there are times the comment came in as an instance of list, sometimes it is not. So to make sure they are all instance of list type, I test it first and ensure that it is a list type whatever the input is, and sort the list according ly. Please run this simple testing code lines:

```
import os
from sphinx_intl import catalog as c
home = os.environ['HOME']
home_base = os.path.join(home, "<where ever the po directory is>"
input_file = os.path.join(home_base, "blender.pot")
output_file = os.path.join(home_base, "test_output_blender.pot")

data = c.load_po(input_file)
c.dump_po(output_file. data, line_width=4096)
```

and load the output file into an editor (ie. vscode) and POEdit to test for python-format errors.